src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmTestResult.java

```diff
@@ -218,7 +223,22 @@ public Boolean isCompleted() {
      * Return a Jenkins build result which matches the result status from AWS Device Farm.
      * @return
      */
-    public Result getBuildResult() {
+    public Result getBuildResult(Boolean ignoreRunError) {
+        if (ignoreRunError != null && ignoreRunError && ExecutionResult.ERRORED.toString().equals(result)) {
+            if (totalCount == skipCount) {
+                result = ExecutionResult.SKIPPED.toString();
```

Comment:
If the static result map stores the enum value, you can skip all of these toString calls here.